native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java

```diff
@@ -435,7 +447,18 @@ public void notifySuccess(Object result) {
                     EntityBodyHandler.closeMessageOutputStream(outputStream);
                     return;
                 }
-                writeContentPart((BMap) result, outputStream);
+                if (result instanceof BError error) {
+                    entity.addNativeData(ENTITY_BYTE_STREAM, null);
+                    EntityBodyHandler.closeMessageOutputStream(outputStream);
+                    throw error;
+                }
+                try {
+                    writeContentPart((BMap) result, outputStream);
+                } catch (Exception e) {
+                    BError error = ErrorCreator.createError(StringUtils.fromString(
```

Comment:
This is an error when writing the stream content right? In that case shall we return an error with the message:
```
Error occurred while writing the stream content: <actual-exception>
```

And throw that error directly rather than calling notify failure